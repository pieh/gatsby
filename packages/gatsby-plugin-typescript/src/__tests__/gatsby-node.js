jest.mock(`../resolve`, () => module => `/resolved/path/${module}`)

const {
  resolvableExtensions,
  onCreateWebpackConfig,
} = require(`../gatsby-node`)
const { tsPresetsFromJsPresets } = require(`../`)
const tsPresetPath = `/resolved/path/@babel/preset-typescript`
const jsOptions = {
  options: {
    presets: [
      [`@babel/preset-env`],
      [`@babel/preset-react`],
      [`@babel/preset-flow`],
    ],
    plugins: [`babel-plugin-remove-graphql-queries`],
  },
  loader: `/resolved/path/babel-loader`,
}

describe(`tsPresetsFromJsPresets`, () => {
  it(`handles empty presets`, () => {
    const presets = []
    expect(tsPresetsFromJsPresets(presets)).toEqual([tsPresetPath])
  })
  it(`replaces preset-flow if it's last`, () => {
    const presets = [`@babel/preset-flow`]
    expect(tsPresetsFromJsPresets(presets)).toEqual([tsPresetPath])
  })
  it(`appends if preset-flow is not last`, () => {
    const presets = [[`@babel/preset-flow`], [`@babel/preset-foo`]]
    expect(tsPresetsFromJsPresets(presets)).toEqual(
      presets.concat([tsPresetPath])
    )
  })
})

describe(`gatsby-plugin-typescript`, () => {
  let args

  beforeEach(() => {
    const actions = {
      setWebpackConfig: jest.fn(),
    }
    const loaders = { js: jest.fn(() => jsOptions) }
    args = { actions, loaders }
  })

  it(`returns correct extensions`, () => {
    expect(resolvableExtensions()).toMatchSnapshot()
  })

  it(`modifies webpack config`, () => {
    const babelConfig = { plugins: [``] }
    const config = {
      loader: jest.fn(),
    }

    onCreateWebpackConfig(
      { config, babelConfig, ...args },
      { compilerOptions: {} }
    )

    expect(args.actions.setWebpackConfig).toHaveBeenCalledTimes(1)
    const lastCall = args.actions.setWebpackConfig.mock.calls.pop()
    expect(lastCall).toMatchSnapshot()
  })
})
