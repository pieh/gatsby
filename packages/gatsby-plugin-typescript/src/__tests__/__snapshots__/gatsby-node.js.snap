// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-typescript modifies webpack config 1`] = `
Array [
  Object {
    "module": Object {
      "rules": Array [
        Object {
          "test": /\\\\\\.tsx\\?\\$/,
          "use": Array [
            Object {
              "loader": "/resolved/path/babel-loader",
              "options": Object {
                "plugins": Array [
                  "babel-plugin-remove-graphql-queries",
                ],
                "presets": Array [
                  Array [
                    "@babel/preset-env",
                  ],
                  Array [
                    "@babel/preset-react",
                  ],
                  "/resolved/path/@babel/preset-typescript",
                ],
              },
            },
          ],
        },
      ],
    },
  },
]
`;

exports[`gatsby-plugin-typescript returns correct extensions 1`] = `
Array [
  ".ts",
  ".tsx",
]
`;
